src/test/java/io/vertx/jdbcclient/JDBCPoolTest.java

```diff
@@ -351,4 +352,40 @@ public void testBatchPreparedStatement(TestContext ctx) {
       }));
     }));
   }
+  
+  @Test
+  public void testPreparedStatementWithBufferParam(TestContext should) {
+    final Async test = should.async();
+    
+    Buffer buffer = Buffer.buffer("Hello world!");
+    
+    client
```

Comment:
this could be improved with instead:

```java
client.query("drop table if exists t")
      .execute()
      .compose(res1 -> client
        .query("create table t (b BLOB)")
        .execute()
        .compose(res2 -> client
          .preparedQuery("insert into t (b) values (?)")
          .execute(Tuple.of(buffer))
          .compose(res3 -> client
            .query("select b from t")
            .execute()
            .onComplete(ctx.asyncAssertSuccess(rows -> {
              should.assertEquals(1, rows.size());
              rows.forEach(row -> {
                Buffer actual = row.getBuffer(0);
                should.assertNotNull(actual);
                should.assertEquals(buffer, actual);
              });
            }))
          )
        )
      );
```

The async object creation can be dropped as `ctx#asyncAssertSuccess` creates an async under the hood.

Reply:
Should good ... done!